frontend/rust-lib/flowy-folder2/src/manager.rs

```diff
@@ -442,21 +442,42 @@ impl FolderManager {
 
   /// Move the view to trash. If the view is the current view, then set the current view to empty.
   /// When the view is moved to trash, all the child views will be moved to trash as well.
+  /// All the favorite views in trash will be handled as well
   #[tracing::instrument(level = "debug", skip(self), err)]
   pub async fn move_view_to_trash(&self, view_id: &str) -> FlowyResult<()> {
     self.with_folder((), |folder| {
-      let view = folder.views.get_view(view_id);
-      folder.add_trash(vec![view_id.to_string()]);
-
-      // notify the parent view that the view is moved to trash
-      send_notification(view_id, FolderNotification::DidMoveViewToTrash)
-        .payload(DeletedViewPB {
-          view_id: view_id.to_string(),
-          index: None,
-        })
-        .send();
+      if let Some(view) = folder.views.get_view(view_id) {
+        let mut all_descendant_views: Vec<Arc<View>> = vec![view.clone()];
+        all_descendant_views.extend(folder.views.get_views_belong_to(view_id));
+
+        let favorite_descendant_views: Vec<ViewPB> = all_descendant_views
+          .iter()
+          .filter(|view| view.is_favorite)
+          .map(|view| view_pb_without_child_views(view.clone()))
+          .collect();
+
+        if !favorite_descendant_views.is_empty() {
+          folder.delete_favorites(
+            favorite_descendant_views
+              .iter()
+              .map(|v| v.id.clone())
+              .collect(),
+          );
+          send_notification("favorite", FolderNotification::DidUnFavoriteView)
+            .payload(RepeatedViewPB {
+              items: favorite_descendant_views,
+            })
+            .send();
+        }
```

Comment:
This could be it's own method, as it clutters the `move_view_to_trash`

Reply:
updated!